src/System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -571,10 +571,17 @@ public override IEnumerable<string> EnumerateDirectories(string path, string sea
         public override IEnumerable<string> EnumerateDirectories(string path, string searchPattern, SearchOption searchOption)
         {
             mockFileDataAccessor.PathVerifier.IsLegalAbsoluteOrRelative(path, "path");
+            var isOriginalPathRooted = mockFileDataAccessor.Path.IsPathRooted(path);
             path = path.TrimSlashes();
             path = mockFileDataAccessor.Path.GetFullPath(path);
-            return GetFilesInternal(mockFileDataAccessor.AllDirectories, path, searchPattern, searchOption)
+            var directories = GetFilesInternal(mockFileDataAccessor.AllDirectories, path, searchPattern, searchOption)
                 .Where(p => !mockFileDataAccessor.StringOperations.Equals(p, path));
+            if (!isOriginalPathRooted)
+            {
+                //if the path was orginally relative, return relative paths: #815
+                directories = directories.Select(p => p.Substring(currentDirectory.Length + 1));
```

Comment:
To rebuild the relative path I tried using `mockFileDataAccessor.Path.GetRelativePath` but it's not available in all frameworks. So I used a substring.

Reply:
I've re-implemented this part by using `Path.Combine` with the original path argument and the "last path component"
I think using a logic similar to `GetRelativePath` won't do the trick.

For instance, if we're in `/foo` where there's one subfolder `/foo/bar`
If we ask to enumerate `.`, the internal enumeration will return `/foo/bar` and getting the path relative to current folder will simply return `bar` not `./bar`.
Same if we ask `../foo`, using relative path would return `bar` and not `../foo/bar`

Basically I take the last part component `bar` which I recombine with whatever was initially given.

Reply:
I don't believe that this is the generally correct approach, see my comment in the test.

I'm wondering whether it would be easier to only fix this for target frameworks where Path.GetRelativePath is available (we are multi-targeting here and could use conditional compilation)